### PR TITLE
The user must specify a category for a new cookbook

### PR DESCRIPTION
**Pull request by thommay**
Bonus fix for a hardcoded URL.

Fixes: #5088, #5092, #5095

### CODE REVIEW

**Comment by thommay** [comment]
This is one of those "did that ever work", @chef/client-core

**Comment by thommay** [comment]
In common with Stove, let's just say "Other" if it's unspecified, since we don't really care anymore

**Comment by lamont-granquist** [comment]
since we don't care could we force everything to "Other" and avoid the rescue/404 logic entirely and null out the command line option?

**Comment by thommay** [comment]
I didn't want to do that just in case people are actually using categories

File: lib/chef/knife/cookbook_site_share.rb [hunk 1]
```diff
@@ -113,19 +113,16 @@ def run
...
+        data["category"]
+      rescue => e
+        if e.response.code == "404"
+          "Other"
+        else
           ui.fatal("Unable to reach Supermarket: #{e.message}. Increase log verbosity (-VV) for more information.")
           Chef::Log.debug("\n#{e.backtrace.join("\n")}")
-          exit(1)
         end
+        exit(1)
```
Comment (coderanger) [inline lib/chef/knife/cookbook_site_share.rb]: This seems like it is in the wrong place, don't we want to return the `"Other"`?

**Comment by lamont-granquist** [comment]
so this is where knife-supermarket got fixed:

https://github.com/chef/knife-supermarket/pull/10/files

it just hardcoded an empty-string category and didn't give a choice.

**Pull request merged**